src/kflags.rs

```diff
@@ -0,0 +1,99 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use bitstruct::bitstruct;
+
+/// The type of system we are booting.
+/// Options include:
+/// * Oxide: A Gimlet or Cosmo; startup information taken from IPCC
+/// * Ethanol-X: An AMD Ethanol-X reference design machine
+/// * Ruby: An AMD Ruby reference design machine
+/// * Grapefruit: A reference design machine with Oxide grapefruit board
+#[derive(Clone, Copy, Debug)]
+#[repr(u8)]
+pub enum Machine {
+    Oxide = 0,
+    EthanolX = 1,
+    Ruby = 2,
+    Grapefruit = 3,
```

Comment:
Grapefruit feels a bit odd to include here. AFAIK it's currently intended to be used with a Ruby system but we there are other possible systems I could see it being used with maybe (we have the 2P titanite in the office for instance).

Reply:
Good point; truthfully, I didn't give it too much thought, and just wanted to put together a list of machines and grapefruit was in my mind.

That said, I know Andy wants to be able to do IPCC over eSPI, which implies some way of knowing that the target machine has grapefruit. But perhaps that becomes a `RubyRed` machine (get it? Ruby Red grapefruit juice? Thanks; I'm here all week).